client/src/lib/components/Buttons/IconButton/AnimatedIconButton.tsx

```diff
@@ -0,0 +1,88 @@
+import React, {useCallback, useEffect, useRef, useState} from 'react';
+import styled from 'styled-components/native';
+import AnimatedLottieView from 'lottie-react-native';
+import {SPACINGS} from '../../../constants/spacings';
+import BaseButton, {BaseButtonProps} from '../BaseButton';
+import {AnimatedIconType} from '../../Icons/AnimatedIcon';
+
+export type BaseIconButtonProps = BaseButtonProps & {
+  noBackground?: boolean;
+  fill?: string;
+};
+
+const StyledIconButton = styled(BaseButton)<BaseIconButtonProps>(props => ({
+  ...(props.noBackground ? {backgroundColor: 'transparent'} : {}),
+  width: props.small ? 36 : 44,
+  height: props.small ? 36 : 44,
+  padding: props.small ? 3 : 7,
+  borderRadius: SPACINGS.SIXTEEN,
+}));
+
+export type IconButtonProps = Omit<BaseIconButtonProps, 'variant'> & {
+  AnimatedIcon: AnimatedIconType;
+  preVariant: BaseButtonProps['variant'];
+  postVariant: BaseButtonProps['variant'];
+};
+
+const AnimatedIconButton: React.FC<IconButtonProps> = ({
+  onPress,
+  preVariant,
+  postVariant,
```

Comment:
Hmm do we have to have a pre and post here? Can't we just react to state changes in an effect on certain props? E.g always animate when variant changes? Now it's like the logic is inside of this component. 

... but I might be missing aspects of the problem.

Reply:
true, variant can be set as a state change, nice